System Check

ig0774 opened the pull request:
I'm posting this PR for comments.

The majority of issues we get seem to be caused by configuration issues, mostly stemming from Sublime running in an environment that users don't expect it to be. This is generally true of OS X as is well-known and at least possible on Ubuntu  (see #655, where the issue is, in essence, that Unity doesn't inherit it's environment from the login shell). In addition, because of the various places configurations can be made (either the `LaTeXTools.sublime-settings` or the project file) it could conceivably be quite hard to track-down which configuration setting is used. In short, we have (and provide) little visibility into the system so that users can understand what is going on and so we can help them troubleshoot their issues.

So I've created a new command which does some basic sanity checks and displays the information in a (hopefully) clear format.

Below is what it currently looks like:

<img width="828" alt="Screen shot showing the output of the command" src="https://cloud.githubusercontent.com/assets/212893/13236393/f620ae2c-d9bb-11e5-941c-d23749163213.png">

There are some differences depending on configuration, i.e., if a program can't be found or executed the status changes to 'missing'. In addition, on MiKTeX, it checks for `texify` instead of `latexmk` and the various environment variables `texify` uses (`BIBTEX`, `LATEX`, etc.).

Short TODO list:
- [x] Verify it works on:
  - [x] Windows 
    - [x] ST2
    - [x] ST3
  - [x] Linux
    - [x] ST2
    - [x] ST3
  - [x] OS X
    - [x] ST2
    - [x] ST3
- [x] Verify changes to `get_settings()` don't break anything
- [x] Add something to kill potentially hanging processes
- [x] Add checks for engine, tex options, tex root if launched from a tex file
- [x] Update everything for new (v3.7.x) code

Are there any other keys bits of information that are missing or would be useful to have? Could anything be done to make the information displayed clearer?

(I've thought about trying to get the actual commands the builder will run, but that turns out to be a major headache to do without actually running the commands or refactoring the builders which may be a possibility down the line.)

r-stein commented:
I have tried it on my windows machine

- `ST3` works and displays the correct results, however it sometimes insert `CR` characters at the end of the line (I think ST buffers always use `LF` only, even with `CRLF` files)
- `ST2` also displays the result and does not have the `CR` problem, but does not support the :white_check_mark: and :x: characters

In addition `biber` is listed as required, although it must be explicitly enabled with texify. Hence I would list it as optional.

ig0774 commented:
@r-stein: Thanks. I've fixed the `CRLF` thing and removed the required column altogether. May have to do the same with the dingbats as I can't get them to work on Windows 7 (I think 8 & 10 have some extra handling for that range of Unicode).

ig0774 commented:
A version of this change has been merged.
